### PR TITLE
Update monitoring docs

### LINKED ISSUE #2143: Update 'monitoring deployed applications' section in reference docs
https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#configuration-monitoring-deployed-applications

With Boot 2.0 support there needs to be a clear separation of how monitoring is working in Boot 1.5.x vs voot 2.0.x.  

Other sections in Monitoring and Management should be reviewed from this perspective as well.

### PR DESCRIPTION
**Pull request by markpollack**
Remove section on spring boot admin
Restructure to reflect monitoring stream/boot 1.x and 2.0 apps.

`./mvnw clean package -DskipTests -P full -pl spring-cloud-dataflow-docs -am` to build

`firefox spring-cloud-dataflow-server-kubernetes-docs/target/generated-docs/index.html` to view

Fixes #2143

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 1]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
-The Actuator library adds HTTP endpoints under the context path `/management` that is also
-a discovery page for available endpoints.  For example, there is a `health` endpoint
-that shows application health information and an `env` that lists properties from
-Spring's `ConfigurableEnvironment`.  By default, only the health and application info
-endpoints are accessible.  The other endpoints are considered to be sensitive
-and need to be http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#production-ready-customizing-endpoints[enabled explicitly via configuration].  If you enable
-sensitive endpoints, you should also
-<<configuration-security,secure the Data Flow server's endpoints>> so that
-information is not inadvertently exposed to unauthenticated users.  The local Data Flow server has security disabled by default, so all actuator endpoints are available.
+The Actuator library adds HTTP endpoints under the context path `/management` which is a discovery page for available managerment endpoints.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: "managerment" -> management

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 2]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
+For example, there is a `health` endpoint that shows application health information and an `env` that lists properties from Spring's `ConfigurableEnvironment`.
+By default, only the health and application info endpoints are accessible.  The other endpoints are considered to be sensitive
+and need to be  https://docs.spring.io/spring-boot/docs/1.5.12.RELEASE/reference/htmlsingle/#production-ready-customizing-endpoints[enabled explicitly via configuration].
+If you enable sensitive endpoints, you should also <<configuration-security,secure the Data Flow server's endpoints>> so that information is not inadvertently exposed to unauthenticated users.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: The formatting here seems a little odd. "<<configuration-security,secure the Data Flow server's endpoints>>"

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 3]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
+The local Data Flow server has security disabled by default, so all actuator endpoints are available.
 
-The Data Flow server requires a relational database, and, if the feature toggled for
-analytics is enabled, a Redis server is also required.  The Data Flow server
-autoconfigures the https://github.com/spring-projects/spring-boot/blob/v1.4.1.RELEASE/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/DataSourceHealthIndicator.java[DataSourceHealthIndicator] and https://github.com/spring-projects/spring-boot/blob/v1.4.1.RELEASE/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/RedisHealthIndicator.java[RedisHealthIndicator] if needed.  The health of these two services is incorporated to the overall health status of the server through the `health` endpoint.
+The Data Flow server requires a relational database, and, if the feature toggled for analytics is enabled, a Redis server is also required.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: "toggled" -> toggle

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 4]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
+The Data Flow server autoconfigures the https://github.com/spring-projects/spring-boot/blob/v1.4.1.RELEASE/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/DataSourceHealthIndicator.java[DataSourceHealthIndicator] and https://github.com/spring-projects/spring-boot/blob/v1.4.1.RELEASE/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/RedisHealthIndicator.java[RedisHealthIndicator] if needed.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: The Boot link point to `v1.4.1.RELEASE` - might be dated content.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 5]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
-and JMX.
+The stream applications deployed by Spring Cloud Data Flow can be based on Spring Boot 1.5 or Spring Boot 2.0.
+ Both versions contains several features for monitoring your application in production.
+However, Spring Boot 1.x and 2.x as well as Spring Cloud Stream 1.x and 2.x differ in how monitoring is implemented.
+Since Spring Cloud Data Flow supports deploying 1.x and 2.x applications, we will cover both cases individually.
 
-A few additional Spring projects provide support for sending
-metrics data to external systems:
+What is common across 1.x and 2.x applications is that Spring Cloud Stream apps can be configured to publish metrics to a messaging middleware destination.
+The https://github.com/spring-cloud/spring-cloud-dataflow-metrics-collector[Spring Cloud Data Flow Metrics Collector] subscribes to this destination and aggregates metrics into a stream based view.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: Maybe the destination property can be included in parenthesis.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 6]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
 
 [[configuration-monitoring-management]]
 == Monitoring and Management
-The Spring Cloud Data Flow server is a Spring Boot application that includes the http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#production-ready[Actuator
+The Spring Cloud Data Flow server is a Spring Boot 1.5 application that includes the  https://docs.spring.io/spring-boot/docs/1.5.12.RELEASE/reference/htmlsingle/#production-ready[Actuator
```
Comment (ghillert) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: We are making a Spring Boot `1.5.12.RELEASE` reference in 4 places. Shouldn't we use properties for that?

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 7]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
+The stream applications deployed by Spring Cloud Data Flow can be based on Spring Boot 1.5 or Spring Boot 2.0.
+ Both versions contains several features for monitoring your application in production.
+However, Spring Boot 1.x and 2.x as well as Spring Cloud Stream 1.x and 2.x differ in how monitoring is implemented.
+Since Spring Cloud Data Flow supports deploying 1.x and 2.x applications, we will cover both cases individually.
 
-A few additional Spring projects provide support for sending
-metrics data to external systems:
+What is common across 1.x and 2.x applications is that Spring Cloud Stream apps can be configured to publish metrics to a messaging middleware destination.
+The https://github.com/spring-cloud/spring-cloud-dataflow-metrics-collector[Spring Cloud Data Flow Metrics Collector] subscribes to this destination and aggregates metrics into a stream based view.
+The Metrics Collector 2.0 server supports collecting metrics from streams that contain only Boot 1.x or 2.x apps as well as streams that contain a mixture of Boot versions.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: Maybe you meant, "not only Boot 1.x ...."

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 8]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
+The Data Flow UI queries the Metrics collector over HTTP to display messages rates next to each deployed application.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: "messages" -> message

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 9]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
+The Metrics Collector can be secured with 'basic' authentication that requires a username and password.
+To set the username and password, use the properties `spring.cloud.dataflow.metrics.collector.username` and `spring.cloud.dataflow.metrics.collector.password` when starting the Data Flow server.
 
-The metrics
-for each application are published when the property `spring.cloud.stream.bindings.applicationMetrics.destination` is set.  This can be set as any other
-application property when deploying an application in Data Flow.  Since
-it is quite common to want all applications in a stream to emit metrics,
-setting it at the Data Flow server level is a good way to achieve that.
+The metrics for each application are published when the property `spring.cloud.stream.bindings.applicationMetrics.destination` is set.
+Using a destination name of `metrics` is a good choice as the Metrics Collector subscribes to that name by default.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: Also, add in the end: ", which of course, can be overridden to be different than the default as needed"

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 10]
```diff
@@ -1199,77 +1156,72 @@ app register --name log --type sink --uri maven://org.springframework.cloud.stre
...
-`/collector/metrics` in JSON format. The Data Flow server accesses this
-endpoint in two distinct ways. The first is by exposing a `/metrics/streams`
-HTTP endpoint that acts as a proxy to the Metrics Collector endpoint. This
-is accessed by the UI when overlaying message rates on the Flow diagrams for
-each stream. It is also accessed to enrich the Data Flow `/runtime/apps`
-endpoint that is exposed in the UI in the `Runtime` tab and in the shell
+The Metrics Collector exposes aggregated metrics under the HTTP endpoint `/collector/metrics` in JSON format.
+The Data Flow server accesses this endpoint in two distinct ways.
+The first is by exposing a `/metrics/streams` HTTP endpoint that acts as a proxy to the Metrics Collector endpoint.
+This is accessed by the UI when overlaying message rates on the Flow diagrams for each stream.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: "Flow diagrams for each stream" -> SCDF's Flo and it's visual representation of streaming pipelines.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 11]
```diff
@@ -1199,77 +1156,72 @@ app register --name log --type sink --uri maven://org.springframework.cloud.stre
...
-http://start-scs.cfapps.io/[Data Flow Initializr] to generate a project and
-then add to the generated Maven pom file the MetricWriter implementation you
-want to use.  The documentation on the Data Flow Metrics project pages provides
-the additional information you need to get started.
+spring.metrics.export.triggers.application.includes=integration**
+----
+
+For 2.x applications, the property is `spring.cloud.stream.metrics.meter-filter` and it does not have a default value, so all metrics are exported.
+
+Note that the Data Flow UI only displays only instantaneous input and output channel message rates.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: 2 "only"s.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 12]
```diff
@@ -1199,77 +1156,72 @@ app register --name log --type sink --uri maven://org.springframework.cloud.stre
...
+Data Flow does not provide its own implementation to store and visualize historical metrics data, instead we integrate with existing monitoring system.
+For Boot 1.x, there is support for sending metrics to the application log and DataDog.
+For Boot 2.x, metrics is backed by the https://micrometer.io/[Micrometer library] that provides a wide range of https://docs.spring.io/spring-boot/docs/2.0.0.RELEASE/reference/htmlsingle/#production-ready-metrics[monitoring systems].
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: Maybe change 2.0.0 to current so it always points to the latest GA release?

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 13]
```diff
@@ -1199,77 +1156,72 @@ app register --name log --type sink --uri maven://org.springframework.cloud.stre
...
+
+
+==== Spring Boot 2.x
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: Maybe we can remove this for 1.5 M1. We can add it to 1.5 RC.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc (hunk 6 above)
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: we didn't for `current` ;) but yea, generally speaking yes.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 14]
```diff
@@ -1083,113 +1083,70 @@ being set to `true`.
...
-Spring Boot lets you http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#production-ready-public-metrics[add your own metrics] to the `/metrics` endpoint either by registering an implementation of the `PublicMetrics` interface or through its integration with http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#production-ready-dropwizard-metrics[Dropwizard].
+=== Monitoring Deployed Stream Applications
 
-The Spring Boot interfaces, `MetricWriter` and `Exporter`, are used to send the
-metrics data to a place where they can be displayed and analyzed. There are
-implementations in Spring Boot to export metrics to Redis, Open TSDB, Statsd,
-and JMX.
+The stream applications deployed by Spring Cloud Data Flow can be based on Spring Boot 1.5 or Spring Boot 2.0.
```
[the remark this answers is not in the document]
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: We support boot 1.0 apps I'd bet....

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc (hunk 6 above)
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: added as part of https://github.com/spring-cloud/spring-cloud-dataflow/issues/2148

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc (hunk 4 above)
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: added to https://github.com/spring-cloud/spring-cloud-dataflow/issues/2148

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc (hunk 13 above)
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: Reworded to indicate it is a WIP.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 15]
```diff
@@ -1199,77 +1156,72 @@ app register --name log --type sink --uri maven://org.springframework.cloud.stre
...
 in the following example:
 
 [source,bash]
 ----
 spring.cloud.stream.metrics.properties=spring.application.name,spring.application.index,spring.cloud.application.*,spring.cloud.dataflow.*
 ----
-Which is the set of application properties values needed to perform aggregation.
-Data Flow also sets the property, as shown `spring.metrics.export.triggers.application.includes` in the following example:
+The values of these keys are used as the tags to perform aggregation.
+In the case of 2.x applications, these key-values map directly onto tags in the micrometer library.
```
[the remark this answers is not in the document]
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: fixed

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 16]
```diff
@@ -1199,77 +1156,72 @@ app register --name log --type sink --uri maven://org.springframework.cloud.stre
...
-then add to the generated Maven pom file the MetricWriter implementation you
-want to use.  The documentation on the Data Flow Metrics project pages provides
-the additional information you need to get started.
+spring.metrics.export.triggers.application.includes=integration**
+----
+
+For 2.x applications, the property is `spring.cloud.stream.metrics.meter-filter` and it does not have a default value, so all metrics are exported.
+
+Note that the Data Flow UI only displays only instantaneous input and output channel message rates.
+Data Flow does not provide its own implementation to store and visualize historical metrics data, instead we integrate with existing monitoring system.
```
[the remark this answers is not in the document]
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: fixed

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 17]
```diff
@@ -1199,77 +1156,72 @@ app register --name log --type sink --uri maven://org.springframework.cloud.stre
...
+For Boot 1.x, there is support for sending metrics to the application log and DataDog.
```
[the remark this answers is not in the document]
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: fixed

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 18]
```diff
@@ -1199,77 +1156,72 @@ app register --name log --type sink --uri maven://org.springframework.cloud.stre
...
+For Boot 2.x, metrics is backed by the https://micrometer.io/[Micrometer library] that provides a wide range of https://docs.spring.io/spring-boot/docs/2.0.0.RELEASE/reference/htmlsingle/#production-ready-metrics[monitoring systems].
+
+
+==== Spring Boot 2.x
+
+We have developed an sample application that show how to modify the `time` and `log` applications and export metrics to InfluxDB using the micrometer library.  A Grafana front end is also provided.
```
[the remark this answers is not in the document]
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: fixed

**Comment by markpollack** [comment]
addressed comments and merged as 66bf8ee58237fd97fb27dff25aa972110e53a324

**Pull request closed without merging**